### PR TITLE
ark restic init-repository: use key data if provided

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <steve@heptio.com>

### CODE REVIEW

File: pkg/cmd/cli/restic/init_repository_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func (f *fakeFactory) Namespace() string {
+	return ""
+}
+
+func TestComplete(t *testing.T) {
+	// no key options provided should error
+	o := &InitRepositoryOptions{}
+	assert.NotNil(t, o.Complete(&fakeFactory{}))
```
Comment (ncdc) [inline pkg/cmd/cli/restic/init_repository_test.go]: Would `assert.Error` work?

File: pkg/cmd/cli/restic/init_repository_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	// both KeyFile and KeyData provided should error
+	o = &InitRepositoryOptions{
+		KeyFile: "/foo",
+		KeyData: "bar",
+	}
+	assert.NotNil(t, o.Complete(&fakeFactory{}))
```
Comment (ncdc) [inline pkg/cmd/cli/restic/init_repository_test.go]: ditto?

File: pkg/cmd/cli/restic/init_repository_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	// TODO test that if KeyFile is provided, the data is used
```
Comment (ncdc) [inline pkg/cmd/cli/restic/init_repository_test.go]: +1

File: pkg/cmd/cli/restic/init_repository_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	// (should move the pkg/restore/FileSystem interface to
+	// a common location and use it here so ioutil.ReadFile
+	// can be easily mocked)
+
+	// if KeyData is provided, it's used
+	o = &InitRepositoryOptions{
+		KeyData: "bar",
+	}
+	assert.Nil(t, o.Complete(&fakeFactory{}))
```
Comment (ncdc) [inline pkg/cmd/cli/restic/init_repository_test.go]: ditto?

**Comment by skriss** [comment]
oh, sure. also gonna restructure the complete code a little. 1 sec

File: pkg/cmd/cli/restic/init_repository_test.go (hunk 3 above)
Comment (skriss) [reply pkg/cmd/cli/restic/init_repository_test.go]: OK with me just doing this refactor as part of this PR?

Comment (ncdc) [reply pkg/cmd/cli/restic/init_repository_test.go]: Yes

**Comment by skriss** [comment]
updated

**Comment by ncdc** [comment]
LGTM

**Pull request merged**